File: tweet-app-backend/src/main/java/com/tweet/security/bo/SignupRequest.java
package com.tweet.security.bo;

import java.io.Serializable;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Signup Request.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SignupRequest implements Serializable {
    private static final long serialVersionUID = -5661633722744629741L;
    private String firstname;
    private String lastname;
    private String username;
    private String dateOfBirth;
    private String emailId;
    private String password;
    private Boolean agreeTerms;
}

File: tweet-app-frontend/src/js/entities/SignupRequest.ts
class SignupRequest {
  firstname: string;
  lastname: string;
  username: string;
  dateOfBirth: string;
  password: string;
  emailId: string;
  agreeTerms: boolean;

  constructor(
    firstname: string,
    lastname: string,
    username: string,
    dateOfBirth: string,
    password: string,
    emailId: string,
    agreeTerms: boolean
  ) {
    this.firstname = firstname;
    this.lastname = lastname;
    this.username = username;
    this.dateOfBirth = dateOfBirth;
    this.password = password;
    this.emailId = emailId;
    this.agreeTerms = agreeTerms;
  }
}

export default SignupRequest;
